tap_postgres/tap.py

```diff
@@ -1,22 +1,127 @@
 """Postgres tap class."""
 
+from pathlib import Path, PurePath
 from typing import List
 
-from singer_sdk import SQLTap, SQLStream
+from singer_sdk import SQLTap
 from singer_sdk import typing as th  # JSON schema typing helpers
-from tap_postgres.client import PostgresStream
+
+from tap_postgres.client import PostgresConnector, PostgresStream
 
 
 class TapPostgres(SQLTap):
-    """Postgres tap class."""
+    """Postgres tap class.
+
+    This class uses a single connector to manage all connections for all of its
+    streams.
+    """
+
     name = "tap-postgres"
     default_stream_class = PostgresStream
 
+    _connector: PostgresConnector = None
+
     config_jsonschema = th.PropertiesList(
         th.Property(
             "sqlalchemy_url",
             th.StringType,
-            required=True,
-            description="Example postgresql://postgres:postgres@localhost:5432/postgres"
+            required=False,
+            description="Full DB connection URL. E.g. 'postgresql://postgres:postgres@localhost:5432/postgres'",
+        ),
+        th.Property(
+            "dialect+driver",
+            th.StringType,
+            required=False,
+            default="postgresql+psycopg2",
+            description="Dialect and driver for database. E.g. 'postgresql+pscyopg2' in 'postgresql+pscyopg2://user:pass@localhost:5432/postgres'",
+        ),
```

Comment:
Could we split these into `dialect` and `driver` as separate inputs, or maybe just remove the config altogether if `postgresql+psycopg2` is a reasonable default for most use cases?

Reply:
I agree I should have pushed back on this more in the `target-postgres` PR I thought it would let folks play around more with SQLAlchemy Dialects but that's really not the goal of these taps/targets and if you really want to overriding in the code yourself is easy.

Reply:
Yeah I'll drop it. If users want to override it, we're letting them pass a sqlalchemy_url instead, so they do it there.